### PR TITLE
Added a troubleshooting section for agent lost

### PR DESCRIPTION
**Pull request by nsuma8989**

### CODE REVIEW

**Comment by nsuma8989** [comment]
@mbelton-buildkite Please can you review this PR

File: pages/agent/v3.md [hunk 1]
```diff
@@ -249,3 +249,15 @@ Exit codes for common signals:
   <td>Termination signal (graceful)</td>
 <tr>
 </table>
+
+## Troubleshooting
```
Comment (mbelton-buildkite) [inline pages/agent/v3.md:253]: Include a blank line after headings:

```suggestion
## Troubleshooting

```

File: pages/agent/v3.md [hunk 2]
```diff
@@ -249,3 +249,15 @@ Exit codes for common signals:
...
+
+Agent after registering to Buildkite API will exchange regular heartbeats to update it is up and running. If Buildkite API does not get any heartbeat requests from an Agent for 10 consecutive minutes that agent will be marked as lost and no further jobs would be assigned to that agent.
+
+There can be various reasons for agent failing to send heartbeat and few of the common reasons are due to networking issues, resource issues such as CPU/Memory/IO on infrastructure where agent is running etc.
+
+In such scenarios along with checking agent logs also look into metrics around networking, cpu, memory and IO to see if you can identify the reason for failure to send heartbeats.
+
+If agents are running on Elastic CI stack with spot instances, sudden termination of spot instances could also lead to agents getting marked as lost. Here is [log collector script](https://github.com/buildkite/elastic-ci-stack-for-aws#collect-logs-via-script) which can be used to collect all the relevant logs from the Elastic CI stack along with metrics to investigate it.
```
Comment (mbelton-buildkite) [inline pages/agent/v3.md:262]: General tightening:

```suggestion
One issue you sometimes need to troubleshoot is when Buildkite loses contact with an agent, resulting in a `-1` exit code. After registering with the Buildkite API, an agent regularly sends heartbeat updates to indicate that it is operational. If the Buildkite API does not receive any heartbeat requests from an agent for 10 consecutive minutes, that agent is marked as lost and will not be assigned any further jobs.

Various factors can cause an agent to fail to send heartbeat updates. Common reasons include networking issues and resource constraints, such as CPU, memory, or I/O limitations on the infrastructure hosting the agent.

In such cases, it's essential to check the agent logs and examine metrics related to networking, CPU, memory, and I/O to help identify the cause of the failed heartbeat updates.

If the agents run on the Elastic CI Stack for AWS with spot instances, the abrupt termination of spot instances can also result in marking agents as lost. To investigate this issue, you can use the [log collector script](https://github.com/buildkite/elastic-ci-stack-for-aws#collect-logs-via-script) script to gather all relevant logs and metrics from the Elastic CI Stack.
```

Please check it still makes sense :)

Comment (nsuma8989) [reply pages/agent/v3.md:262]: This looks good. Thank you for the suggestion. I updated it

**Review by mbelton-buildkite — APPROVED** [review]

**Pull request merged**